org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/guice/AbstractGuiceAwareExecutableExtensionFactory.java

```diff
@@ -49,15 +50,35 @@ public Object create() throws CoreException {
 		try {
 			final Class<?> clazz = getBundle().loadClass(clazzName);
 			final Injector injector = getInjector();
+			if (injector == null) {
+				throw handleCreationError(null);
+			}
 			final Object result = injector.getInstance(clazz);
 			if (result instanceof IExecutableExtension)
 				((IExecutableExtension) result).setInitializationData(config, null, null);
 			return result;
+		} catch (Throwable e) {
```

Comment:
I have just looked into the changes. You might consider two remarks:
 - I would not catch `Throwable` but `Exception`.
 - I would make `handleCreationError` protected.